src/main/java/de/tum/in/www1/artemis/service/connectors/localvc/LocalVCServletService.java

```diff
@@ -288,14 +296,183 @@ else if (exception instanceof LocalVCForbiddenException) {
     }
 
     /**
-     * Call the local CI system to process the new push there.
+     * Create a submission, trigger the respective build, and process the results.
      *
      * @param commitHash the hash of the last commit.
      * @param repository the remote repository which was pushed to.
-     * @throws LocalCIException        if something goes wrong preparing the queueing of the build job.
-     * @throws VersionControlException if the commit belongs to the wrong branch (i.e. not the default branch of the participation).
+     * @throws ContinuousIntegrationException if something goes wrong with the CI configuration.
+     * @throws VersionControlException        if the commit belongs to the wrong branch (i.e. not the default branch of the participation).
      */
     public void processNewPush(String commitHash, Repository repository) {
-        localCIConnectorService.orElseThrow().processNewPush(commitHash, repository);
+        long timeNanoStart = System.nanoTime();
```

Comment:
I would recommend splitting this method in several submethods for better readability, as this method is very hard to read. Each submethod could perform one specific step.

Reply:
This applies to most of the following methods, too.